### ISSUE TITLE
Ability to create Custom roles in mangle UI

### ISSUE THREAD

**Issue by sowjanya-eks** [vmware/mangle#136]
Hello Mangle Team, (@rpraveen-vmware, @ranjansvmware)

Thank you for joining yesterday's call. As we discussed here is the new request for custom role.

In order to create the endpoints, it requires admin role. Typically we need a custom role where we can choose the privilege's like add, edit and read, write capabilities (such as creating endpoints, injecting chaos etc.)

Regards,
Sowjanya

**Comment by sowjanya-eks** [comment]
Could anyone please acknowledge this issue. @rpraveen-vmware, @ranjansvmware

**Comment by ranjansvmware** [comment]
I guess the ask is:
1. Create role of persona as: Allow fault injection, hiding or not allowing add/update/delete endpoints, no access to config settings.
2. Tenancy support: User should able to customise its own resources and not others, unless resource is marked as shared explicitly. Eg: User should be able to add/update/delete it own added endpoint.

**Comment by sowjanya-eks** [comment]
Yes, you're understanding is correct. Is it feasible for you to implement in next release? @rpraveen-vmware, @ranjansvmware

**Comment by sowjanya-eks** [comment]
This is a feature request to support custom role where we can choose the individual privilege's.
